src/simple_cache_client.rs

```diff
@@ -55,6 +55,24 @@ impl MomentoRequest for &str {
     }
 }
 
+impl MomentoRequest for &[u8] {
+    fn into_bytes(self) -> Vec<u8> {
+        self.to_owned()
+    }
+}
+
+impl MomentoRequest for &Vec<u8> {
+    fn into_bytes(self) -> Vec<u8> {
+        self.clone().into_bytes()
+    }
+}
+
+impl MomentoRequest for &String {
+    fn into_bytes(self) -> Vec<u8> {
+        self.clone().into_bytes()
+    }
+}
```

Comment:
Instead of defining "Into"-like traits that hide expensive payload copies from & to standard types, should we just redefined our trait to use the standard definitions for type conversions?
```rust
pub trait MomentoRequest : Into<Vec<u8>> {}
impl <T> MomentoRequest for T where T : Into<Vec<u8>> {} // I'm not sure if the blanket impl is actually needed on this one
```

That would let you delete like 20 lines =) and would make it more clear the intent that this is a from/into thing. If a user wants to clone before passing in to us that's always an option, but in Rust we should be clear about what we need at the call site - which is a wholly owned `Vec<u8>` or something that can be converted `Into` it.

Reply:
Also, the way this is used, both existing and in your new change, is not a "request" it's a "MomentoBytes" or just "momento::Bytes." It needs to be renamed...

Reply:
I'm fine with this. The `MomentoRequest` trait has been a misnomer since I started looking at the code, uh, yesterday. I was just trying to follow along with the existing code.

I'll update the trait to be called `IntoBytes` and change it's uses to suit.